common/src/main/java/com/redhat/cloud/notifications/models/BehaviorGroup.java

```diff
@@ -43,13 +44,16 @@
  * See https://in.relation.to/2016/08/04/introducing-distinct-pass-through-query-hint/ for more details about that hint.
  */
 // TODO Move this query back to BehaviorGroupResources when hints are implemented in Mutiny.Query.
-@NamedQuery(
-        name = "findByBundleId",
-        query = "SELECT DISTINCT b FROM BehaviorGroup b LEFT JOIN FETCH b.actions a " +
-                "WHERE (b.accountId = :accountId OR b.accountId IS NULL) AND b.bundle.id = :bundleId " +
-                "ORDER BY b.created DESC, a.position ASC",
-        hints = @QueryHint(name = BehaviorGroup.HINT_PASS_DISTINCT_THROUGH, value = "false")
-)
+@NamedQueries({
+        @NamedQuery(
+                name = "findByBundleId",
+                query = "SELECT DISTINCT b FROM BehaviorGroup b LEFT JOIN FETCH b.actions a " +
+                        "WHERE (b.accountId = :accountId OR b.accountId IS NULL) AND b.bundle.id = :bundleId " +
+                        "ORDER BY b.created DESC, a.position ASC",
+                hints = @QueryHint(name = BehaviorGroup.HINT_PASS_DISTINCT_THROUGH, value = "false")),
+        @NamedQuery(name = "findByDisplayName",
+                query = "SELECT b FROM BehaviorGroup b WHERE b.displayName = :displayName"),
```

Comment:
This isn't wrong, but I'm not really sure if there is a different between using a named query or directly using the ORM to filter by a column.

I'm guessing we should stick to one and only use the named queries for specific cases.

@gwenneg WDYT?

Reply:
Also, this doesn't take into account the accountId. 

We should prevent from having the same (accountId,DisplayName). Note that accountId could be `null` and needs to be considered in the query (dynamic query) because `b.accountId = null` won't work well (`b.accountId is NULL)`

Reply:
I like NamedQueries because they are "close" to the class where they belong to. So the SQL query isn't located somewhere in the code. IMO it's a bit like getters, setters or even business methods. They should be located in the class whose behavior they are changing instead of being somewhere in other classes or so.
But that's just my personal opinion :)

Reply:
Another use case would be: if you search for "something which works with the behavior group feature" you would probably take a look at the behavior group first and find the queries immediately.

Reply:
I think putting all queries in a similar place saves dev time.

We can't make all notifications queries named because of the many dynamic queries we already have. In the meantime, we can put all queries in the `*Resources`/`*Repository` classes.

So I think named queries aren't the best productivity option, at least for me.  I will always look for a query in the `*Resources`/`*Repository` class first before realizing the query is actually named.

Reply:
In general we do have the same opinion about it :) 

In Spring Boot the Repositories are a good place to put the queries in. Quarkus / JPA probably doesn't have that feature, so *Resources/*Repository classes are something like Quarkus / JPA version of it.
I would also be fine with it.

Reply:
> Could you please add a test for this?

There is one. BehaviorGroupResourcesTest.shouldThrowExceptionWhenBehaviorGroupWithAccountIdAndDisplayNameAlreadyExists()

Reply:
@gwenneg I will open a new clean PR which moves the NamedQueries back in the BehaviorGroupResources.

Reply:
> @gwenneg I will open a new clean PR which moves the NamedQueries back in the BehaviorGroupResources.

Why not in this one?

Reply:
Thought it would be a bit out of scope as it's more or less a result task of the reactiveness removal.

I can also do it here. Give me some minutes.

Reply:
I don't want to deal with merge conflicts in the evening :/

Reply:
it's part of this PR since commit no. [7d1e79f](https://github.com/RedHatInsights/notifications-backend/pull/1027/commits/7d1e79faa2307f77d960707f66ced7fdb537f17b)

Reply:
My question was only focused on the named query you added. The other one has to stay anyway for now.

Reply:
added again :D

Reply:
done.